include/xgboost/c_api.h

```diff
@@ -1386,4 +1388,118 @@ XGB_DLL int XGBoosterFeatureScore(BoosterHandle handle, const char *json_config,
                                   bst_ulong *out_dim,
                                   bst_ulong const **out_shape,
                                   float const **out_scores);
+
+/*!
+ * \brief Initialize the collective communicator.
```

Comment:
Can we mark this as experimental? I see that we might be making breaking changes in the future as we might try to unify some other features like CPU/GPU communicator interface, or clean up the input parameters.